tasks/transform_conformance/src/lib.rs

```diff
@@ -266,4 +317,113 @@ impl TestCase {
+            .split('/')
+            .collect::<Vec<&str>>()
+            .join("-");
+
+        let mut target_path = target_root.join("fixtures").join(new_file_name);
+        target_path.set_extension("test.js");
+        let content = format!(
+            r#"
+                import {{expect, test}} from 'bun:test';
+                test("exec", () => {{
+                    {content}
+                }})
+            "#
+        );
+        fs::write(&target_path, content).unwrap();
+        let source_text = fs::read_to_string(&target_path).unwrap();
+        let source_type = SourceType::from_path(&target_path).unwrap();
+        let transformed_program =
+            Parser::new(&allocator, &source_text, source_type).parse().program;
+        let result =
+            Codegen::<false>::new(source_text.len(), CodegenOptions).build(&transformed_program);
+
+        fs::write(&target_path, result).unwrap();
+
+        target_path
+    }
+
+    fn transform(&self, path: &Path) -> String {
+        let allocator = Allocator::default();
+        let source_text = fs::read_to_string(path).unwrap();
+        let source_type = SourceType::from_path(path).unwrap();
+        let transformed_program =
+            Parser::new(&allocator, &source_text, source_type).parse().program;
+
+        let semantic =
+            SemanticBuilder::new(&source_text, source_type).build(&transformed_program).semantic;
+        let (symbols, scopes) = semantic.into_symbol_table_and_scope_tree();
+        let symbols = Rc::new(RefCell::new(symbols));
+        let scopes = Rc::new(RefCell::new(scopes));
+        let transformed_program = allocator.alloc(transformed_program);
+
+        Transformer::new(&allocator, source_type, &symbols, &scopes, self.transform_options())
+            .build(transformed_program);
+        Codegen::<false>::new(source_text.len(), CodegenOptions).build(transformed_program)
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum TestCaseKind {
+    Transform,
+    Exec,
+}
+
+impl TestCaseKind {
+    fn from_path(path: &Path) -> Option<Self> {
+        // in `exec` directory
+        if path.parent().is_some_and(|path| path.file_name().is_some_and(|n| n == "exec"))
+            && path.extension().is_some_and(|ext| VALID_EXTENSIONS.contains(&ext.to_str().unwrap()))
+        {
+            return Some(Self::Exec);
+        }
+        // named `exec.[ext]`
+        if path.file_stem().is_some_and(|name| name == "exec")
+            && path.extension().is_some_and(|ext| VALID_EXTENSIONS.contains(&ext.to_str().unwrap()))
+        {
+            return Some(Self::Exec);
+        }
+
+        // named `input.[ext]``
+        if path.file_stem().is_some_and(|name| name == "input")
+            && path.extension().is_some_and(|ext| VALID_EXTENSIONS.contains(&ext.to_str().unwrap()))
+        {
+            return Some(Self::Transform);
+        }
+
+        None
+    }
+}
+
+struct SnapshotOption {
```

Comment:
Move this to somewhere higher? It's weird to have a struct at the bottom of the file. Maybe it's time to split this file into some smaller files.